gap/grahom.gi

```diff
@@ -872,3 +872,104 @@ function(D, t)
   n := DigraphNrVertices(D);
   return IsDigraphColouring(D, ImageListOfTransformation(t, n));
 end);
+
+InstallMethod(MaximalCommonSubdigraph, "for a pair of digraphs",
+[IsDigraph, IsDigraph],
+function(D1, D2)
+  local B, out1, out2, adj, good, p, Clqus, M, l,
+        embedding1, embedding2, iso;
+
+  iso := IsomorphismDigraphs(D1, D2);
+
+  if iso <> fail then
+    return [DigraphCopy(D1), IdentityTransformation, AsTransformation(iso)];
+  fi;
+
+  out1 := OutNeighbours(D1);
+  out2 := OutNeighbours(D2);
+
+  adj := function(pair)
```

Comment:
I'm pretty sure that we've discussed this already, but I think you could improve the performance here, by:

1. Not using `Cartesian` but instead doing:
~~~~
L := List([1 .. Size(out1) * Size(out2)], x -> []);
for i in [1 .. Size(out1)] do
  for j in [1 .. Size(out2)] do
    if isadj(i, j) then
       Add(L[i], j); 
    fi;
  od;
od;
~~~~

Also the function `isadj` could be improved, by to run through the out neighbours of `D1` and `D2` once by using the adjacency matrix of `D1` and `D2`